### DISCUSSION TITLE
OpenVSCode-Server Offline

### DISCUSSION THREAD

**Discussion by spdavern** [gitpod-io/openvscode-server#383] [Q&A] [answered]
I'm running openvscode-server (v 1.68.0) on an offline (no internet connection) Linux machine in a docker container.  Much of the functionality seems to be unavailable because vscode-cdn.net is not accessible:
- image preview (e.g. jpg, png) doesn't work
- markdown preview doesn't work
- Git Graph functionality doesn't work

From troubleshooting on systems with online access, in the case of image preview, it seems the viewer (main.js and main.css) is downloaded from vscode-cdn.net for every image previewed. Not only does this seem inefficient, I was hoping I would be able to build the docker image with everything it needs to be fully functional in the offline environment.  How do I do that?  Is it even possible?  From Discussion #335  I'm concerned openvscode might be a dead end.

**Comment by jeanp413** [comment, marked as answer]
Hi @spdavern, sadly is not possible anymore with openvscode-server as it will require too many changes that will end up in merge conflicts

**Reply by spdavern** [reply]
It is truly a sad day.  Thank you for your rapid, clear response!
